tests/Unit/CMManagementPoint.tests.ps1

```diff
@@ -0,0 +1,482 @@
+[System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '')]
+param ()
+
+$script:dscModuleName   = 'ConfigMgrCBDsc'
+$script:dscResourceName = 'DSC_CMManagementPoint'
+
+function Invoke-TestSetup
+{
+    try
+    {
+        Import-Module -Name DscResource.Test -Force -ErrorAction 'Stop'
+    }
+    catch [System.IO.FileNotFoundException]
+    {
+        throw 'DscResource.Test module dependency not found. Please run ".\build.ps1 -Tasks build" first.'
+    }
+
+    $script:testEnvironment = Initialize-TestEnvironment `
+        -DSCModuleName $script:dscModuleName `
+        -DSCResourceName $script:dscResourceName `
+        -ResourceType 'Mof' `
+        -TestType 'Unit'
+
+    # Import Stub function
+    $script:moduleRoot = Split-Path -Parent (Split-Path -Parent $PSScriptRoot)
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath 'Stubs\ConfigMgrCBDscStub.psm1') -Force -WarningAction SilentlyContinue
+}
+
+function Invoke-TestCleanup
+{
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+# Begin Testing
+try
+{
+    InModuleScope $script:dscResourceName {
+        $moduleResourceName = 'ConfigMgrCBDsc - DSC_CMManagementPoint'
+
+        $getInput = @{
+            SiteCode       = 'Lab'
+            SiteServerName = 'MP.contoso.com'
+        }
+
+        $getMpReturnNoSQL = @{
```

Comment:
Is this a valid config for a box to not have SQL installed or not point to an SQL box?
Question is to know if I should do additonal tests.

Reply:
SQL isn't required to specify a SQL server, connection will work, just will not actually connect to anything